src/App.js

```diff
@@ -3,20 +3,24 @@ import {
     BrowserRouter as Router,
     Switch,
     Route,
-    Link,
 } from 'react-router-dom';
 
 import './App.css';
 import Home from './Home.js';
 import QuizController from './QuizController.js';
 
+import IntervalExercise, { generateRandomInterval } from './IntervalExercise.js'
+
 function App() {
     return (
         <Router>
             <div className="app">
                 <Switch>
                     <Route path="/Quiz">
-                        <QuizController />
+                        <QuizController
+                            exerciseClass={IntervalExercise}
```

Comment:
Curious if there's a good way not to have to thread these two properties through the QuizController, which doesn't use them directly?

Reply:
The alternative option would be to allow `QuizController` to take `Quiz` as a child and then render it. It would look like this:

```jsx
<QuizController>
    <Quiz exerciseClass={IntervalExercise}  questionGenerator={generateRandomInterval} />
</QuizController>
```

It's not clear to me this would actually be better and it might be more complicated bubbling information up. In my opinion passing these through is fine.

Reply:
Another, similar option, would be to use the React [higher-order component](https://reactjs.org/docs/higher-order-components.html) pattern whereby you write a function that returns a wrapped component. That would look like this in the quiz component file:

```jsx
export default controlledQuiz(Quiz)
```

and then you could just use the

```jsx
<Quiz exerciseClass={IntervalExercise}  questionGenerator={generateRandomInterval} />
````

component on its own inside the `Route`. But to be honest, the HOC pattern is not my favorite (I personally find it a little confusing) and it seems like overkill here.